packages/client/hmi-client/src/utils/logger.ts

```diff
@@ -0,0 +1,75 @@
+// HMI-Client logging service
+import API from '@/api/api';
+import { createLogger, LogEvent, LoggerHook, LogLevel } from 'vue-logger-plugin';
+// TODO: add logic for different modes
+// const env = process.env.NODE_ENV;
+
+// LogDetails Interface
+interface LogDetails {
+	level: LogLevel;
+	message: string;
+}
+
+// Usage:
+// import logger from '@/utils/logger';
+class LogBuffer {
+	logs: LogDetails[];
+
+	constructor() {
+		this.logs = [];
+	}
+
+	clearLogs = () => {
+		this.logs = [];
+	};
+
+	add = (log: LogDetails) => {
+		this.logs.push(log);
+	};
+
+	isEmpty = (): boolean => {
+		if (this.logs.length > 0) {
+			return false;
+		}
+		return true;
+	};
+
+	getLogBuffer = (): LogDetails[] => this.logs;
+
+	startService = () => {
+		setInterval(() => {
+			this.sendLogsToServer();
+		}, 5000);
+	};
+
+	sendLogsToServer = async () => {
+		if (!this.isEmpty()) {
+			await API.post(`/logs/`, {
+				logs: this.getLogBuffer()
+			})
+				.then(() => {
+					this.clearLogs();
+				})
+				.catch((error) => {
+					console.error(error);
+				});
```

Comment:
Would prefer if we can stick with either async/await-style, or promise-style.  But not both at the same time as it makes it a bit harder to understand.

I think the code here, as-is,  it technically works but the await is redundant.

Reply:
fixed.